src/io/pti.jl

```diff
@@ -943,4 +943,744 @@ function _populate_defaults!(data::Dict)
+    
+    return sub_data
+end
+
+
+"""
+Parses PM generator data to PSS(R)E-style.
+"""
+function _pm2psse_generator(pm_gen::Dict{String, Any},area::Int64, owner::Int64, zone::Int64)
+    sub_data = Dict{String, Any}()
+    sub_data["I"] = pm_gen["gen_bus"] # Not default allowed
+    sub_data["ID"] = "\'$(pm_gen["source_id"][end])\'"
+    sub_data["STAT"] = get(pm_gen, "gen_status", _default_generator["STAT"])
+    sub_data["PG"] = get(pm_gen, "pg", _default_generator["PG"])
+    sub_data["QG"] = get(pm_gen, "qg", _default_generator["QG"])
+    sub_data["QT"] = get(pm_gen, "qmax", _default_generator["QT"])
+    sub_data["QB"] = get(pm_gen, "qmin", _default_generator["QB"])
+    sub_data["VS"] = get(pm_gen, "vg", _default_generator["VS"])
+    sub_data["MBASE"] = get(pm_gen, "mbase", _default_generator["MBASE"])
+    sub_data["PT"] = get(pm_gen, "pmax", _default_generator["PT"])
+    sub_data["PB"] = get(pm_gen, "pmin", _default_generator["PB"])
+    sub_data["AREA"] = get(pm_gen, "area", area)
+    sub_data["OWNER"] = get(pm_gen, "owner", owner)
+    sub_data["ZONE"] = get(pm_gen, "zone", zone)
+    sub_data["O1"] = get(pm_gen, "o1", owner)
+    
+    _export_remaining!(sub_data, pm_gen, _pti_defaults["GENERATOR"])
+    
+    return sub_data
+end
+
+
+"""
+Parses PM branch data to PSS(R) E-style.
+"""
+function _pm2psse_branch(pm_br::Dict{String, Any}, owner::Int64)
+    sub_data = Dict{String, Any}()
+    sub_data["I"] = pm_br["f_bus"] # Not default allowed
+    sub_data["J"] = pm_br["t_bus"] # Not default allowed
+    sub_data["CKT"] = "\'$(pm_br["source_id"][end])\'"
+    sub_data["R"] = pm_br["br_r"]
+    sub_data["X"] = pm_br["br_x"]
+    sub_data["B"] = 0. # or (b_fr + b_to)/2
+    sub_data["RATEA"] = get(pm_br, "rate_a", _default_branch["RATEA"])
+    sub_data["RATEB"] = get(pm_br, "rate_b", _default_branch["RATEB"])
+    sub_data["RATEC"] = get(pm_br, "rate_c", _default_branch["RATEC"])
+    sub_data["GI"] = get(pm_br, "g_fr", _default_branch["GI"])
+    sub_data["BI"] = get(pm_br, "b_fr", _default_branch["BI"])
+    sub_data["GJ"] = get(pm_br, "g_to", _default_branch["GJ"])
+    sub_data["BJ"] = get(pm_br, "b_to", _default_branch["BJ"])
+    sub_data["ST"] = get(pm_br, "br_status", _default_branch["ST"])
+    sub_data["O1"] = get(pm_br, "o1", owner)
+
+    _export_remaining!(sub_data, pm_br, _pti_defaults["BRANCH"])
+
+    return sub_data
+end
+
+
+"""
+Parses PM transformer branch to PSS(R) E-style.
+returns a dict with all the keys
+later pass this dict to _print_pti_str with differents _transformer_dtypes
+
+Reference: PSSE 33 - POM - 5-20
+"""
+function _pm2psse_2w_tran(pm_br::Dict{String, Any}, owner::Int64, sbase::Real, source="pti")
+
+    sub_data = Dict{String, Any}()
+    
+    # TRANSFORMER FIRST LINE PARAMETERS
+    sub_data["I"] = pm_br["f_bus"]
+    sub_data["J"] = pm_br["t_bus"]
+    sub_data["K"] = 0
+    sub_data["CKT"] = source == "pti" ? "\'$(pm_br["source_id"][5])\'" : "\'$(pm_br["source_id"][end])\'"
```

Comment:
Same note as above, `"source_id"` may not be defined.

Reply:
Solved